### PR TITLE
Add api test

### PR DESCRIPTION
**Pull request by arufanov**
Add api test

### CODE REVIEW

**Comment by arufanov** [comment]
@yurichervoniy 
* how to insert data into db according to spring philosopy?

**Review by yurichervoniy — APPROVED** [review]

File: src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(TenantsApiController.class)
```
[the remark this answers is not in the document]
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java:16]: @yurichervoniy 
* please provide example

[the remark this answers is not in the document]
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java]: @yurichervoniy ]
fixed, check please

File: src/test/java/com/a5lab/tabr/AbstractControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.a5lab.tabr.domain.tenants.TenantsApiController;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/AbstractControllerTests.java:11]: @RunWith is an old annotation from JUnit 4 to use test runners.

@WebMvcTest already has @ExtendWith({SpringExtension.class})

So pls remove this old annotation

File: src/test/java/com/a5lab/tabr/AbstractControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+@WebMvcTest(TenantsApiController.class)
+public abstract class AbstractControllerTests {
+  @Autowired
+  public MockMvc mockMvc;
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/AbstractControllerTests.java:15]: It's not critical but if you remove @RunWith
also make mockMvc protected like this
protected MockMvc mockMvc

File: src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    Mockito.when(tenantService.findAll()).thenReturn(tenantList);
+
+    // How to seed database for all tests
+    String content = mockMvc.perform(get("/api/v1/tenants")
+            .contentType(APPLICATION_JSON)
+        )
+        .andExpect(status().isOk())
+        .andReturn()
+        .getResponse()
+        .getContentAsString();
```
Comment (yurichervoniy) [inline src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java:32]: I think it's better to check, for example
 .andExpect(jsonPath("$").isArray())
        .andExpect(jsonPath("$", hasSize(tenantList.size())))
        .andExpect(jsonPath("$[0].id", equalTo(-1)))
        .andExpect(jsonPath("$[0].name", equalTo("firstName")))
        or
        .andExpect(jsonPath("$[*].id", containsInAnyOrder(-123, -2534, -456)))
        .andExpect(jsonPath("$[*].name", hasItems("test123", "fdgdsgf", "vnlclboc")));

File: src/test/java/com/a5lab/tabr/AbstractControllerTests.java (hunk 2 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/AbstractControllerTests.java]: @yurichervoniy 
* fixed it.

File: src/test/java/com/a5lab/tabr/AbstractControllerTests.java (hunk 3 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/AbstractControllerTests.java]: @yurichervoniy 
fixed it.

File: src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java (hunk 4 above)
Comment (arufanov) [reply src/test/java/com/a5lab/tabr/domain/tenants/TenantsApiControllerTests.java:32]: @yurichervoniy 
* thanks, fixed it.

File: src/test/java/com/a5lab/tabr/AbstractControllerTests.java (hunk 2 above)
Comment (yurichervoniy) [reply src/test/java/com/a5lab/tabr/AbstractControllerTests.java:11]: It's very strange. Can we use @ExtendWith(SpringExtension.class) instead of Runwith?

Comment (yurichervoniy) [reply src/test/java/com/a5lab/tabr/AbstractControllerTests.java:11]: You can resolve this comment

**Review by yurichervoniy — APPROVED** [review]

**Pull request merged**